src/matcher/index.ts

```diff
@@ -8,10 +8,17 @@ const evaluateMatcher = (
   if (isMatcherLeaf(matcher)) {
     const { match, info } = matcher;
     const notMatch = (module: string) => !match(module);
+    const newMatched = remaining.filter(match);
+    const newRemaining = remaining.filter(notMatch);
     return {
-      matched: remaining.filter(match),
-      remaining: remaining.filter(notMatch),
-      infos: typeof info === "undefined" ? [] : [info],
+      matched: newMatched,
+      remaining: newRemaining,
+      infos:
```

Comment:
A bug was identified during testing (see https://github.com/wei2912/planwithus-lib/pull/30/commits/c7275257a04d322ce320b42ef17bef277bb1b2ec#diff-087eaa394c3ca994402686b95b6f2b70fcc0da151b3dcbc58717a20577ee759fR48-R68) where a triggered matcher would always add on an `info` string regardless of whether they matched any module or not.